resources/repo.rb

```diff
@@ -0,0 +1,92 @@
+property :repo_name, String, name_attribute: true
+property :repo_type, String, default: 'maven-hosted'
+property :config, Hash
+property :api_url, String
+property :api_user, String
+property :api_password, String
+
+def apiclient
+  @apiclient ||= Nexus3::Api.new(api_url, api_user, api_password)
+end
+
+load_current_value do
+  nexus_api 'list_repo' do
```

Comment:
You can't use a resource here, because you need the result of the API call to feed the `current_value`